Penteract/Assets/Scripts/TriggerDialogueSwapCharacters.cpp

```diff
@@ -1,29 +1,29 @@
-#include "TriggerSwapCharacters.h"
+#include "TriggerDialogueSwapCharacters.h"
 
 #include "GameplaySystems.h"
 #include "GameObject.h"
 
 #include "DialogueManager.h"
 
-EXPOSE_MEMBERS(TriggerSwapCharacters) {
+EXPOSE_MEMBERS(TriggerDialogueSwapCharacters) {
     MEMBER(MemberType::GAME_OBJECT_UID, gameControllerUID)
 };
 
-GENERATE_BODY_IMPL(TriggerSwapCharacters);
+GENERATE_BODY_IMPL(TriggerDialogueSwapCharacters);
 
-void TriggerSwapCharacters::Start() {
+void TriggerDialogueSwapCharacters::Start() {
     gameController = GameplaySystems::GetGameObject(gameControllerUID);
     if(gameController) dialogueManagerScript = GET_SCRIPT(gameController, DialogueManager);
 }
 
-void TriggerSwapCharacters::Update() {
+void TriggerDialogueSwapCharacters::Update() {
 	
 }
 
-void TriggerSwapCharacters::OnCollision(GameObject& /*collidedWith*/, float3 /*collisionNormal*/, float3 /*penetrationDistance*/, void* /*particle*/) {
+void TriggerDialogueSwapCharacters::OnCollision(GameObject& /*collidedWith*/, float3 /*collisionNormal*/, float3 /*penetrationDistance*/, void* /*particle*/) {
     
     if (dialogueManagerScript) {
-        dialogueManagerScript->SetActiveDialogue(&dialogueManagerScript->dialoguesArray[9]);
+        dialogueManagerScript->SetActiveDialogue(&dialogueManagerScript->dialoguesArray[18]);
```

Comment:
If there's no difference between this and TriggerDialogueFinalLvl1.cpp, make those classes a generic class with an exposed variable as the index. Also, it would be nice if you checked the size before accessing it to avoid Out of range exceptions.